### PR TITLE
Log process id

### PR DESCRIPTION
**Pull request by GTANAdam**
I don't think this needs any description

### CODE REVIEW

**Comment by VertiHydro** [comment]
Nice idea. This could be really useful.

**Comment by larryhohack** [comment]
nice work

**Comment by madMAx43v3r** [comment]
I can already see all the complaints about it not compiling on everybodys machine, can you surround that with an `#ifdef` to make sure the function and the header exists?

**Comment by GTANAdam** [comment]
I haven't tested on Windows yet, but the API seems to point that it returns a DWORD, so it must work.

File: src/chia_plot.cpp [hunk 1]
```diff
@@ -16,6 +16,17 @@
...
 #include <iostream>
 
+#ifdef __linux__ 
+	#include <unistd.h>
+	#define GETPID getpid
+#elif _WIN32
+	#include <processthreadsapi.h>
+	#define GETPID GetCurrentProcessId
+#else
+	// Possibly Darwin here
```
Comment (madMAx43v3r) [inline src/chia_plot.cpp:26]: Should define a dummy `GETPID` here then.

File: src/chia_plot.cpp [hunk 2]
```diff
@@ -16,6 +16,17 @@
...
 #include <iostream>
 
+#ifdef __linux__ 
+	#include <unistd.h>
+	#define GETPID getpid
+#elif _WIN32
+	#include <processthreadsapi.h>
+	#define GETPID GetCurrentProcessId
+#else
+	#define GETPID
```
Comment (madMAx43v3r) [inline src/chia_plot.cpp:26]: that won't compile, how about: `#define GETPID() int(-1)`

Comment (GTANAdam) [reply src/chia_plot.cpp]: My apologies, I haven't tested it so I had no idea about that.

**Pull request merged**